src/options/utils/search.js

```diff
@@ -0,0 +1,90 @@
+export function createSearchRules(search) {
+  const tagLists = [[], []];
+  /**
+   * @type Array<{
+   *   scope: string;
+   *   pattern: string | RegExp;
+   *   negative: boolean;
+   * }>
+   */
+  const rules = [];
+  search = search.toLowerCase();
+  let offset = 0;
+  while (search[offset] === ' ') offset += 1;
+  while (offset < search.length) {
+    const negative = search[offset] === '!';
+    if (negative) offset += 1;
+    let prefix =
+      search.slice(offset).match(/^(#|(?:name|code)(?:\+re)?:)/)?.[1] || '';
+    if (prefix) offset += prefix.length;
+    const quote =
+      (!prefix || prefix.endsWith(':')) &&
+      search.slice(offset).match(/^['"]/)?.[0];
+    if (quote) offset += 1;
+    let pattern = '';
+    const endChar = quote || ' ';
+    while (offset < search.length) {
+      const ch = search[offset];
+      if (quote && ch === quote && search[offset + 1] === quote) {
+        // escape quotes by double it
+        pattern += quote;
+        offset += 2;
+      } else if (ch !== endChar) {
+        pattern += ch;
+        offset += 1;
+      } else {
+        break;
+      }
+    }
+    if (quote) {
+      if (offset < search.length) offset += 1;
+      else throw new Error('Unmatched quotes');
+    }
+    if (prefix === '#') {
+      tagLists[+negative].push(pattern);
+    } else {
+      // Strip ':'
+      prefix = prefix.slice(0, -1);
+      if (prefix.endsWith('+re')) {
+        rules.push({
+          scope: prefix.slice(0, -3),
+          pattern: new RegExp(pattern, 'i'),
+          negative,
+        });
+      } else {
+        const reMatches = pattern.match(/^\/(.*?)\/(\w*)$/);
```

Comment:
I think this should be the first thing to try, otherwise it won't match when the entire input is `/foo #bar/`

Reply:
This is intended. I don't want to match regexp with spaces this way, as it is ambiguous and confusing.
So `/foo #bar/` will be parsed into string `/foo` and tag `bar/`. To input a regexp we can use `re:"foo #bar"` or `/foo\s#bar/`.

Reply:
Strictly speaking the new behavior is correct, but it's not really useful because tags aren't likely to end in `/`...

Reply:
Yeah, by right they should be two strings since `/` is not allowed in tags. But an invalid tag matches nothing so it does not hurt. I think it might be a bad idea for `#...` to be sometimes a tag and sometimes not.